spark_pipeline_framework/transformers/http_data_receiver/v4/http_data_receiver.py

```diff
@@ -0,0 +1,277 @@
+from pyspark.rdd import RDD
+from pyspark.sql.functions import col, from_json
+from pyspark.sql.types import (
+    StructField,
+    StringType,
+    StructType,
+    MapType,
+    IntegerType,
+    BooleanType,
+)
+
+from spark_pipeline_framework.transformers.http_data_receiver.v4.http_data_receiver_processor import (
+    HttpDataReceiverProcessor,
+)
+from spark_pipeline_framework.transformers.http_data_receiver.v4.schema import (
+    REQUEST_GENERATOR_TYPE,
+    RESPONSE_PROCESSOR_TYPE,
+)
+from spark_pipeline_framework.utilities.capture_parameters import capture_parameters
+from pyspark.ml.param import Param
+from pyspark.sql.dataframe import DataFrame
+from spark_pipeline_framework.logger.yarn_logger import get_logger
+from spark_pipeline_framework.progress_logger.progress_log_metric import (
+    ProgressLogMetric,
+)
+from spark_pipeline_framework.progress_logger.progress_logger import ProgressLogger
+from spark_pipeline_framework.transformers.framework_transformer.v1.framework_transformer import (
+    FrameworkTransformer,
+)
+from spark_pipeline_framework.utilities.oauth2_helpers.v2.oauth2_client_credentials_flow import (
+    OAuth2Credentails,
+)
+from spark_pipeline_framework.utilities.pipeline_helper import TransformerMixin
+from spark_pipeline_framework.utilities.spark_data_frame_helpers import (
+    create_empty_dataframe,
+)
+
+
+class HttpDataReceiver(FrameworkTransformer, TransformerMixin):
+    """
+    This is a generic class to call a http api and return the response
+    """
+
+    # noinspection PyUnusedLocal
+    @capture_parameters
+    def __init__(
+        self,
+        name: str,
+        view_name: str,
+        error_view: str,
+        http_request_generator: REQUEST_GENERATOR_TYPE,
+        response_processor: RESPONSE_PROCESSOR_TYPE,
+        batch_count: Optional[int] = None,
+        batch_size: Optional[int] = None,
+        cache_storage_level: Optional[StorageLevel] = None,
+        credentials: Optional[OAuth2Credentails] = None,
+        auth_url: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        run_sync: bool = False,
+        raise_error: bool = False,
+        progress_logger: Optional[ProgressLogger] = None,
+    ) -> None:
+        """
+        Transformer to call and receive data from an API
+
+
+        :param http_request_generator: Generator to build next http request
+        :param view_name: name of the view to read the response into
+        :param name: name of transformer
+        :param parameters: parameters
+        :param progress_logger: progress logger
+        it supposed to return a HelixHttpRequest to be used to call the API or return None to end the API call loop
+        :param response_processor: it can change the result before loading to spark df
+        :param error_view: (Optional) log the details of the api failure into `error_view` view.
+        :param raise_error: (Optional) Raise error in case of api failure
```

Comment:
sequence of params in incorrect. please correct
Some params are missing as well like run_sunc

Reply:
progress_logger is still missing